src/emulator/storage/files.ts

```diff
@@ -291,11 +295,28 @@ export class StorageLayer {
     const file = new StoredFile(finalMetadata, filePath);
     this._files.set(filePath, file);
 
+    return { upload, file };
+  }
+
+  public persistUpload(finalizedUpload: FinalizedUpload): void {
+    const { upload, file } = finalizedUpload;
+    const filePath = this.path(upload.bucketId, upload.objectId);
+
     this._persistence.deleteFile(filePath, true);
     this._persistence.renameFile(upload.fileLocation, filePath);
 
     this._cloudFunctions.dispatch("finalize", new CloudStorageObjectMetadata(file.metadata));
-    return { upload: upload, file: file };
+  }
+
+  public finalizeUpload(uploadId: string): FinalizedUpload | undefined {
```

Comment:
Can we simplify these methods somehow? Having three methods named `finishUpload`, `persistUpload`, and `finalizeUpload` violate the Easy to Use, Hard to Misuse principle, https://medium.com/@kralka/easy-to-use-hard-to-misuse-cb0f1410ad72

Reply:
Done.